expression/builtin_compare_vec.go

```diff
@@ -615,11 +617,146 @@ func (b *builtinNullEQStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var err error
+	if err = b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	return VecCompareInt(mysql.HasUnsignedFlag(b.args[0].GetType().Flag), mysql.HasUnsignedFlag(b.args[1].GetType().Flag), result, buf, resOfLT, result)
+}
+
+// VecCompareInt compares two integers.
+func VecCompareInt(isUnsigned0, isUnsigned1 bool, lColumn, rColumn *chunk.Column, f func(val int64, isNull bool, err error) (int64, bool, error), result *chunk.Column) error {
```

Comment:
Maybe we can make `resOfLT`, `compareNull`, `CompareUint64` and `CompareInt64` be vectorized?
It will bring better performance.

Reply:
I've thought about this before . When these functions are vectorized, each vectorized function has a loop. I'm worried that it is really bring better performance?

Reply:
Maybe we can make a benchmark to test it?

Reply:
ok, i will create another PR to make resOfLT, compareNull, CompareUint64 and CompareInt64 be vectorized, and compare the performance.

Reply:
I had created a new PR: https://github.com/pingcap/tidb/pull/12710。
It seems that the new PR is faster then this PR. Is it because the  "branch prediction"?

Reply:
I guess 2 reasons:
1. branch prediction
2. There are too many function calls in this implementation, and it's not friendly to the cache.

Reply:
This PR can be closed?